### PR TITLE
MODLD-342: Derive marc code from the value provided in https://bibframe/vocab/lite/link property

### PR DESCRIPTION
**Pull request by pkjacob**
….me/vocab/lite/link property

### CODE REVIEW

File: src/main/java/org/folio/linked/data/mapper/dto/monograph/common/CategoryMapperUnit.java [hunk 1]
```diff
@@ -35,18 +36,25 @@ public abstract class CategoryMapperUnit implements SingleResourceMapperUnit {
 
   private final CoreMapper coreMapper;
   private final HashService hashService;
-  private final BiConsumer<Category, Object> categoryConsumer;
```
Comment (pkjacob) [inline src/main/java/org/folio/linked/data/mapper/dto/monograph/common/CategoryMapperUnit.java:38]: Removed this `BiConsumer` and changed it to an abstract method `addToDestination`

File: src/test/java/org/folio/linked/data/mapper/dto/monograph/work/sub/CarrierMapperUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  private CarrierMapperUnit carrierMapperUnit;
+  @Autowired
+  private ObjectMapper objectMapper;
+  @MockBean
+  private HashService hashService;
+
+  @Test
+  void toEntity_shouldDeriveMarcCodeFromLink() {
+    // given
+    String expectedMarcCode = "ha";
```
Comment (askhat-abishev) [inline src/test/java/org/folio/linked/data/mapper/dto/monograph/work/sub/CarrierMapperUnitTest.java:36]: var (here and in the remaining tests)

File: src/main/java/org/folio/linked/data/mapper/dto/monograph/instance/sub/CarrierMapperUnit.java [hunk 3]
```diff
@@ -16,14 +16,24 @@
...
-    }, CATEGORY);
+    super(coreMapper, hashService, CATEGORY);
+  }
+
+  @Override
+  protected void addToParent(Category category, Object parentDto) {
+    if (parentDto instanceof Instance instance) {
+      instance.addCarrierItem(category);
+    }
+    if (parentDto instanceof InstanceReference instance) {
```
Comment (askhat-abishev) [inline src/main/java/org/folio/linked/data/mapper/dto/monograph/instance/sub/CarrierMapperUnit.java:30]: maybe better to name variable as ```instanceReference``` (same for ```workReference```)
I've noticed that same variable name causes some confusion

**Review by askhat-abishev — APPROVED** [review]

**Review by PBobylev — APPROVED** [review]

File: src/test/java/org/folio/linked/data/mapper/dto/monograph/work/sub/CarrierMapperUnitTest.java (hunk 2 above)
Comment (pkjacob) [reply src/test/java/org/folio/linked/data/mapper/dto/monograph/work/sub/CarrierMapperUnitTest.java]: Done

**Pull request merged**